src/Extensions/ExtensionManifest.re

```diff
@@ -51,118 +51,39 @@ module Decode = {
-                                                                )
-                                                                |> default(
-                                                                    false,
-                                                                   )
-                                                                >>= (
-                                                                  enableProposedApi =>
-                                                                    succeed({
-                                                                    name,
-                                                                    version,
-                                                                    author,
-                                                                    displayName,
-                                                                    description,
-                                                                    main,
-                                                                    icon,
-                                                                    categories,
-                                                                    keywords,
-                                                                    engines,
-                                                                    activationEvents,
-                                                                    extensionDependencies,
-                                                                    extensionPack,
-                                                                    extensionKind,
-                                                                    contributes,
-                                                                    enableProposedApi,
-                                                                    })
-                                                                )
-                                                            )
-                                                        )
-                                                    )
-                                                )
-                                            )
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    )
-                )
-            )
-        )
+    Json.Decode.(
+      obj(({field, whatever, _}) =>
+        {
+          name: field.required("name", string),
+          version: field.required("version", string),
+          author:
+            whatever(
+              one_of([
+                ("author", field.monadic("author", author)),
+                ("publisher", field.monadic("publisher", string)),
+                ("default", succeed("Unknown Author")),
+              ]),
+            ),
```

Comment:
This is one of the shortcomings of the `obj` decoder. Because the `field.*` decoders aren't "normal" monadic decoders, they can't be composed. So we need an escape hatch. This is it, called `whatever` while I think of a better name for it.